### PR TITLE
Add Kubecost SSP AddOn docs

### PR DESCRIPTION
**Pull request by elenalape**
*Issue #, if available:*

https://github.com/aws-samples/ssp-eks-patterns/pull/10#issuecomment-1005220527 as requested by @shapirov103.

*Description of changes:*

Added usage instructions for the Kubecost AddOn. Project itself lives [here](https://github.com/kubecost/kubecost-ssp-addon).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/addons/kubecost.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+```
+
+## `KubecostAddOn` Options (props)
+
+#### `namespace: string` (optional)
+
+The namespace where Kubecost will be installed. Defaults to `kubecost`.
+
+#### `kubecostToken: string` (optional)
```
Comment (shapirov103) [inline docs/addons/kubecost.md:44]: Some customers may object to token being part of their source code. Let's think of a solution that leverages CSI Secrets Store driver or any other Kubernetes native approach (including straight k8s secrets) to pass it over.

File: docs/addons/kubecost.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import * as cdk from '@aws-cdk/core';
+import { EksBlueprint } from '@aws-quickstart/ssp-amazon-eks';
+import { KubecostAddOn } from '@kubecost/kubecost-ssp-addon';
+
+
+export default class KubecostConstruct extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string) {
+        super(scope, id);
+        // AddOns for the cluster
+        // Include the Kubecost AddOn in your list
```
Comment (shapirov103) [inline docs/addons/kubecost.md:25]: minor: let's drop "in your list" since it is passed to the builder directly.

File: docs/addons/index.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ The framework currently supports the following add-ons.
 | [`ClusterAutoscalerAddOn`](./cluster-autoscaler) | Adds the standard cluster autoscaler ([Karpenter](https://github.com/awslabs/karpenter) is coming)|
 | [`ContainerInsightsAddOn`](./container-insights) | Adds Container Insights support integrating monitoring with CloudWatch |
 | [`CoreDnsAddOn`](./coredns.md) | Adds CoreDNS Amazon EKS add-on. CoreDNS is a flexible, extensible DNS server that can serve as the Kubernetes cluster DNS |
+| [`Kubecost`](./kubecost.md) | Adds [Kubecost](https://kubecost.com) cost analyzer to the EKS cluster |
```
Comment (shapirov103) [inline docs/addons/index.md:20]: Minor: for consistency with the rest let's change `Kubecost` to `KubecostAddOn` in the first column.

**Review by shapirov103 — COMMENTED** [review]
Nice work! A few small comments.

File: docs/addons/kubecost.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+####  `values?: { [key: string]: any }` (optional)
+
+Custom values to pass to the chart. Config options: https://github.com/kubecost/cost-analyzer-helm-chart/blob/master/README.md#config-options 
+
+#### `customPrometheus: string` (optional)
+
+Kubecost comes bundled with a Prometheus installation. However, if you wish to integrate with an external Prometheus deployment, provide your local Prometheus service address with this format `http://..svc`.
+Note: integrating with an existing Prometheus is only officially supported under Kubecost paid plans and requires some extra configurations on your Prometheus: https://docs.kubecost.com/custom-prom.html
+
+#### `installPrometheusNodeExporter: string` (optional)
```
[the remark this answers is not in the document]
Comment (elenalape) [reply docs/addons/kubecost.md:61]: indeed 😅

File: docs/addons/kubecost.md (hunk 1 above)
Comment (elenalape) [reply docs/addons/kubecost.md:44]: I've removed it from the docs, but kept the functionality in — it is an optional parameter.

**Comment by elenalape** [comment]
Thank you @shapirov103! Just pushed the suggested changes.

File: docs/addons/kubecost.md (hunk 1 above)
Comment (shapirov103) [reply docs/addons/kubecost.md]: Is it a token that is sensitive in nature as far repercussions of falling into the wrong hands?
I can provide an approach how to leverage the CSI driver and AWS Secret manager for integration shortly if it helps. It creates a very visible precedent and others will be tempted to use the same approach for credentials, etc.
Do you think your customers will be open to use it by passing directly in the code?

**Review by shapirov103 — COMMENTED** [review]
Let's clarify the secret management part and we should be good.

**Review by shapirov103 — APPROVED** [review]
lgtm

**Pull request merged**